core/src/HIP/Kokkos_HIP_ZeroMemset.cpp

```diff
@@ -28,7 +28,7 @@ namespace Impl {
 void zero_with_hip_kernel(const HIP& exec_space, void* dst, size_t cnt) {
   Kokkos::parallel_for(
       "Kokkos::ZeroMemset via parallel_for",
-      Kokkos::RangePolicy<Kokkos::HIP>(exec_space, 0, cnt),
+      Kokkos::RangePolicy<Kokkos::HIP, size_t>(exec_space, 0, cnt),
```

Comment:
Since `cnt`'s type is `size_t`, it actually makes sense that the policy's index type is `size_t`, right ?

Reply:
yeah, otherwise the size check of the index in `RangePolicy` starts throwing errors